Makefile

```diff
@@ -1,3 +1,20 @@
+help:
+	@echo "Usage: make <target>"
+	@echo "Targets:"
+	@echo "  all - install, format, lint, test, build"
+	@echo "  install - install dependencies"
+	@echo "  upgrade - upgrade dependencies"
+	@echo "  format - format code"
+	@echo "  lint - lint code"
+	@echo "  test - run tests"
+	@echo "  build - build package"
+.PHONY: help
+
+
+all: install format lint test build
```

Comment:
I'm not sure I love this one.  What's the use-case for this?  I could see it replacing the action in linting and testing where these are all separate actions, but I don't think I've ever wanted to do all the actions at once.